pkg/apis/policy/types.go

```diff
@@ -51,6 +51,19 @@ type PodDisruptionBudgetStatus struct {
 
 	// total number of pods counted by this disruption budget
 	ExpectedPods int32 `json:"expectedPods"`
+
+	// DisruptedPod contains information about pods whose eviction was
+	// processed by the API server eviction subresource handler but has not
+	// yet been observed by the PodDisruptionBudget controller.
+	// A pod will be in this map from the time when the API server starts
+	// deleting the pod to the time when PodDisruptionBudget controller notices
+	// DeletionTimestamp set on the pod. The key in the map is the name of the pod
+	// and the value is the time when the deletion was supposed to strat. If
+	// the deletion didn't occur and a pod is still there it will be removed from
+	// the list automatically by PodDisruptionBudget controller after some time.
+	// If everything goes smooth this map should be empty for the most of the time.
+	// Large number of entries in the map may indicate problems with pod deletions.
+	DisruptedPods map[string]unversioned.Time `json:"disruptedPods" protobuf:"bytes,5,rep,name=disruptedPods"`
```

Comment:
It is not 2000 pods per cluster. It is 2000 pods per single disruption budget. And anyway, allowing >2000 disruptions is a bit weird, I guess that anyone running >2000 pods would either want to have limited number of disruptions (say 30) or don't care about disruptions at all because of the vast amount of pods. And if by any chance they care then they probably can deal with a bit slower deletion.
And ,btw, in alpha we had only 1 disruption at a time.